docs/howto/operator/gcp/dataflow.rst

```diff
@@ -0,0 +1,180 @@
+
+.. _howto/operator:DataflowCreateJavaJobOperator:
+.. _howto/operator:DataflowCreatePythonJobOperator:
+
+Starting a new job
+""""""""""""""""""
+
+To create a new pipeline using the source file (JAR in Java or Python file) use
+the create job operators. The source file can be located on GCS or on the local filesystem.
+:class:`~airflow.providers.google.cloud.operators.dataflow.DataflowCreateJavaJobOperator`
+or
+:class:`~airflow.providers.google.cloud.operators.dataflow.DataflowCreatePythonJobOperator`
+
+Please see the notes below on Java and Python specific SDKs as they each have their own set
+of execution options when running pipelines.
+
+Here is an example of creating and running a pipeline in Java:
+
+.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_dataflow.py
+    :language: python
+    :dedent: 4
+    :start-after: [START howto_operator_start_java_job]
+    :end-before: [END howto_operator_start_java_job]
+
+.. _howto/operator:DataflowTemplatedJobStartOperator:
+
+Templated jobs
+""""""""""""""
+
+Templates give the ability to stage a pipeline on Cloud Storage and run it from there. This
+provides flexibility in the development workflow as it separates the development of a pipeline
+from the staging and execution steps. To start a templated job use the
+:class:`~airflow.providers.google.cloud.operators.dataflow.DataflowTemplatedJobStartOperator`
+
+.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_dataflow.py
+    :language: python
+    :dedent: 4
+    :start-after: [START howto_operator_start_template_job]
+    :end-before: [END howto_operator_start_template_job]
+
+See the `list of Google-provided templates that can be used with this operator
+<https://cloud.google.com/dataflow/docs/guides/templates/provided-templates>`_.
+
+Execution options for pipelines
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Dataflow has multiple options of executing pipelines. It can be done in the following modes:
+asynchronously (fire and forget), blocking (wait until completion), or streaming (run indefinitely).
+In Airflow it is best to use asynchronous pipelines as blocking ones tax the Airflow resources by listening
+to the job until it completes.
+
+Asynchronous execution
+""""""""""""""""""""""
+
+Dataflow jobs are by default asynchronous - however this is dependent on the application code (contained in the JAR
+or Python file) and how it is written. In order for the Dataflow job to execute asynchronously, ensure the
+pipeline objects are not being waited upon (not calling ``waitUntilFinish`` or ``wait_until_finish`` on the
+``PipelineResult`` in your application code).
+
+This is the recommended way to execute your pipelines when using Airflow.
+
+Use the Dataflow monitoring or command-line interface to view the details of your pipeline's results after Airflow
+runs the operator.
+
+Blocking execution
```

Comment:
Asynchronous execution was added in https://github.com/apache/airflow/pull/11726